src/main/scala/scalismo/mesh/boundingSpheres/PointSetDistance.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2015 University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package scalismo.mesh.boundingSpheres
+
+import breeze.numerics._
+import BSDistance.{Distance2, Index, _}
+import scalismo.geometry.{Point, Vector, _3D}
+import scalismo.mesh.TriangleMesh
+
+
+trait PointSetDistance {
+  def closestPoint(point: Point[_3D]): (Point[_3D], Double, Int)
+}
+
+
+/**
+  * PointSetDistance
+  */
+object PointSetDistance {
```

Comment:
I would be in favor of naming this VantagePointTree instead of PointSetDistance which is anyway confusing as one expects a distance to be returned when calling the factory methods. If we want a generic name instead of the algorithm' one, then I would be in favor of SpatialIndex